File: Day2/Solution.py
class Solution(object):
    def fizzBuzz(self, n):
        """
        :type n: int
        :rtype: List[str]
        """
        l = []
        for i in range(1, n + 1):
            if (i % 3 == 0) and (i % 5 == 0):
                l.append("FizzBuzz")
            elif i % 3 == 0:
                l.append("Fizz")
            elif i % 5 == 0:
                l.append("Buzz")
            else:
                l.append(str(i))
        return l

def main():
    sol = Solution()
    
    n = 15
    result = sol.fizzBuzz(n)
    print(f"FizzBuzz results for n = {n}:")
    print(result)

if __name__ == "__main__":
    main()

File: Day2/Solution.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<String> fizzBuzz(int n) {
        List<String> l = new ArrayList<>();

        for (int i = 1; i <= n; i++) {
            if (i % 15 == 0) {
                l.add("FizzBuzz");
            } else if (i % 3 == 0) {
                l.add("Fizz");
            } else if (i % 5 == 0) {
                l.add("Buzz");
            } else {
                l.add(Integer.toString(i));
            }
        }
        return l;
    }

    public static void main(String[] args) {
        Solution sol = new Solution();

        int n = 15;
        List<String> result = sol.fizzBuzz(n);

        // Print the whole list at once
        System.out.println(result);
    }
}
